src/components/captions-control/captions-control-mini.tsx

```diff
@@ -0,0 +1,112 @@
+import {createPortal, useEffect} from 'preact/compat';
+import {registerToBottomBar} from '../bottom-bar';
+import {redux, ReservedPresetNames, style} from '../../index';
+import {Icon, IconType} from '../icon';
+import {h, VNode} from 'preact';
+import {ButtonControl} from '../button-control';
+import {Tooltip} from '../tooltip';
+import {Button} from '../button';
+import {CVAAOverlay} from '../cvaa-overlay';
+import {useState} from 'preact/hooks';
+import {connect} from 'react-redux';
+import {Text} from 'preact-i18n';
+import {CaptionsMenu} from '../captions-menu';
+import {SmartContainer} from '../smart-container';
+import {CaptionsControl} from './captions-control';
+
+const mapStateToProps = state => ({
+  controlsToMove: state.bottomBar.controlsToMove
+});
+
+const COMPONENT_NAME = 'CaptionsControlMini';
+const CaptionsControlMini = connect(mapStateToProps)((props, context) => {
+  const [shouldRender, setShouldRender] = useState(false);
+
+  const {player} = context;
+  const {controlsToMove} = props;
+  const targetId: HTMLDivElement | Document = (document.getElementById(player.config.targetId) as HTMLDivElement) || document;
+  const portalSelector = `.overlay-portal`;
+  let removeOverlay: any = null;
+
+  useEffect(() => {
+    registerToBottomBar(CaptionsControl.displayName, player, () => registerComponent());
+  }, []);
+
+  const registerComponent = (): any => {
+    return {
+      ariaLabel: props.captionsLabelText,
+      displayName: COMPONENT_NAME,
+      order: 5,
+      svgIcon: {type: IconType.ClosedCaptionsOn},
+      onClick: () => onButtonClick(),
+      component: () => {
+        return getComp(props);
+      },
+      selfManagement: true
+    };
+  };
+
+  useEffect(() => {
+    setShouldRender(controlsToMove.includes(CaptionsControl.displayName));
+  }, [controlsToMove]);
+
+  const getIsTextOnFromStore = (): boolean => {
```

Comment:
i feel like in a functional component there's probably a nicer way to wrap up the getter of this value
maybe with useMemo